### PR TITLE
Furniture Charcoal smoker pseudo item

### LINKED ISSUE #41435: Charcoal smoker can only load 200 charcoal
# Describe the bug
the furniture charchoal smoker can now only load 200 charges instead of "infinite". Which makes smoking 20 l of stuff impossible (needed 500). If setting anything higher than 200, it won't reload it.

# Steps To Reproduce
Have a charcoal smoker, load it with charcoal, check charges.

# Expected behavior
Loading at least 500 charcoal

# Screenshots
grafik

# Versions and configuration
- OS: Windows
    - OS Version: 10.0 1903
- Game Version: 0.E-3369-gc738dee [64-bit]
- Graphics Version: Tiles
- Game Language: English [en]
- Mods loaded: [
    Dark Days Ahead [dda],
    Graphical Overmap [Graphical_Overmap],
    Nonperishable Overhaul [Nonperishable_Overhaul],
    Fewer Farms [Fewer_Farms]
]

# Additional context

### PR DESCRIPTION
**Pull request by Maleclypse**
#### Summary

SUMMARY: None

#### Purpose of change

Fixes #41435

### CODE REVIEW

**Comment by dannyisdude** [comment]
Shouldn't it allow 2000 charges, because that is what the charcoal smoker previously allowed?

**Comment by Maleclypse** [comment]
> Shouldn't it allow 2000 charges, because that is what the charcoal smoker previously allowed?

Sure thing. I only could remember it held more than 500 that's referenced in the issue referenced.

**Comment by dannyisdude** [comment]
I had literally just written up a fix for this myself and was about to submit it before seeing you already had 10 min earlier. lol
Though I will say, you could have used less lines of JSON if you implemented it similarly to the char_forge_pseudo item.

**Comment by Maleclypse** [comment]
> I had literally just written up a fix for this myself and was about to submit it before seeing you already had 10 min earlier. lol
> Though I will say, you could have used less lines of JSON if you implemented it similarly to the char_forge_pseudo item.
> Either way, yay for the fix!

I'm happy for anyone to edit and fix my jsons :) I am ready to move back to my other PR's.  or if you have a better fix I'm also good with closing this :)  no stress. let me know

**Comment by dannyisdude** [comment]
Realized my fix seems to have an error anyway. 
You forgot to change the pseudo item for f_smoking_rack_active which is under the entry for f_smoking_rack.

**Comment by dannyisdude** [comment]
Seem that in order to craft, you have to add the new pseudo item to every crafting recipe that requires a charcoal smoker. Or maybe there's a better way to do it idk. 
But either way, this fix makes it so you can longer craft with the charcoal smoker again.

**Comment by dannyisdude** [comment]
Also, might wanna rename the item to Smoking Rack, so it doesn't show up in the crafting requirements as charcoal smoker.

File: data/json/items/tool/cooking.json [hunk 1]
```diff
@@ -146,6 +146,20 @@
     "max_charges": 200,
     "flags": [ "ALLOWS_REMOTE_USE" ]
   },
+  {
+    "id": "f_char_smoker",
```
Comment (ZhilkinSerg) [inline data/json/items/tool/cooking.json:150]: Please use `pseudo_` prefix or `_pseudo` suffix.

File: data/json/items/tool/cooking.json [hunk 2]
```diff
@@ -146,6 +146,20 @@
...
+    "name": { "str": "smoking rack" },
+    "description": "This is a crafting_pseudo_item if you have it something is wrong.",
+    "material": [ "steel" ],
+    "symbol": ";",
+    "color": "light_gray",
+    "ammo": [ "charcoal" ],
+    "pocket_data": [ { "pocket_type": "MAGAZINE", "ammo_restriction": { "charcoal": 2000 } } ],
+    "sub": "char_smoker",
+    "max_charges": 2000,
+    "flags": [ "ALLOWS_REMOTE_USE" ]
```
Comment (ZhilkinSerg) [inline data/json/items/tool/cooking.json:161]: It should probably have `"PSEUDO"` flag too?

Comment (Maleclypse) [reply data/json/items/tool/cooking.json]: Thanks on both :)

**Comment by dannyisdude** [comment]
It looks like you still need to change furniture-tools.json to reflect the change from f_char_smoker to pseudo char smoker.

File: data/json/items/tool/cooking.json (hunk 2 above)
Comment (KorGgenT) [reply data/json/items/tool/cooking.json]: it technically won't change anything, as furniture pseudo items add the pseudo flag in form_from_map

**Comment by Night-Pryanik** [comment]
We have separate json file for pseudo items like this. Maybe better place your item there?

**Comment by Maleclypse** [comment]
> We have separate json file for pseudo items like this. Maybe better place your item there?

I thought we did but for some reason, probably distraction,  I wasn't having luck finding it.  Thanks I'll try and find it to move it.

**Comment by dannyisdude** [comment]
> > We have separate json file for pseudo items like this. Maybe better place your item there?
> 
> I thought we did but for some reason, probably distraction, I wasn't having luck finding it. Thanks I'll try and find it to move it.

json/items/basecamp.json

**Comment by kevingranade** [comment]
data/json/items/fake.json

**Pull request merged**